lib/http-client.js

```diff
@@ -36,4 +35,35 @@ module.exports = class {
       };
     });
   }
+
+  defaultHeaders() {
+    const headers = { Accept: 'application/json+fhir' };
+    if (this.authorizationHeader) headers.Authorization = this.authorizationHeader;
+    return headers;
+  }
+
+  async request(httpVerb, url, ...rest)  {
+    const params = Array.from(rest || []);
+    // Configuration options are always the last param when they are present
+    const configIndex = params.length - 1;
+
+    // axios calls look like:
+    // - httpVerb(url, optionalConfig)
+    // - httpVerb(url, optionalBody, optionalConfig)
+    // If we didn't receive any optional arguments, then we can just add the
+    // headers as the last argument.
+    // If we did receive an optional argument, we need to determine whether it
+    // is a request body or configuration. The only configuration argument we
+    // use is headers, which is not a FHIR field, so if the last optional
+    // argument contains a headers field, it is a configuration and not a body.
+    // If it is a configuration object, we need to merge the headers.
+    // If it's a body, we can add the headers as the last object.
+    if (params.length > 0 && params[configIndex].headers) {
```

Comment:
```
let config = params.pop()

if (config && config.headers)
```
would this work?

Reply:
You have to account for when you do have `config`, but don't have headers, so that leads to this:
```
    let config = params.pop();

    if (config) {
      if (config.headers) {
        const newHeaders = Object.assign({}, this.defaultHeaders(), config.headers);
        config = Object.assign({}, config, { headers: newHeaders });
      } else {
        params.push(config);
        config = { headers: this.defaultHeaders() };
      }
    } else {
      config = { headers: this.defaultHeaders() };
    }

    params.push(config);
    return await this.axiosInstance[httpVerb](url, ...params);
```